### DISCUSSION TITLE
UnhandledRejection error when trying to use metals

### DISCUSSION THREAD

**Discussion by MakonnenMak** [scalameta/coc-metals#400] [Q&A] [answered]
Hey all. Recently switched computers and followed the same exact readme to install metals again, but I'm running into some issues.

Installed Coc using Vim-Plug
`Plug 'neoclide/coc.nvim', {'branch': 'release'}`

Using Nvim and installed metals using  
`:CocInstall coc-metals`

CocConfig
```
 {
           "metals.javaHome": "/usr/bin/java"
 }
```

CocInfo output
```
## versions                                                                                    
vim version: NVIM v0.4.4                                                                       
node version: v16.2.0                                                                         
coc.nvim version: 0.0.80-825a54d780                                                              
coc.nvim directory: /home/makonnen/.vim/plugged/coc.nvim                                          
term: xterm-kitty                                                                                 
platform: linux 
                                                                                 
## Log of coc.nvim                                                                              
2021-05-31T18:43:55.074 INFO (pid:6696) [plugin] - coc.nvim 0.0.80-825a54d780 initialized with nod│~
2021-05-31T18:44:00.911 INFO (pid:6696) [attach] - receive notification: installExtensions []     │~
2021-05-31T18:44:04.122 INFO (pid:6696) [attach] - receive notification: showInfo []

```

When opening a `.scala` file I get this error message
```
[coc.nvim]: UnhandledRejection: spawn ENOTDIR
Error: spawn ENOTDIR
    at ChildProcess.spawn (node:internal/child_process:415:11)
    at Object.spawn (node:child_process:609:9)
    at Object.spawn (/home/makonnen/.config/coc/extensions/node_modules/coc-metals/node_modules/promisif
y-child-process/index.cjs:134:46)
    at Object.fetchMetals (/home/makonnen/.config/coc/extensions/node_modules/coc-metals/node_modules/me
tals-languageclient/lib/fetchMetals.js:7:38)
    at k (/home/makonnen/.config/coc/extensions/node_modules/coc-metals/lib/index.js:28:2623)
    at /home/makonnen/.config/coc/extensions/node_modules/coc-metals/lib/index.js:28:1666
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Ran into this before, but rerunning the installation fixed it for me from this [thread](https://github.com/scalameta/coc-metals/issues/255). That's not working right now so was curious if anyone had any clues as to why this is happening or some good starting points to resolving the issue.

**Comment by ckipp01** [comment, marked as answer]
Thanks for the report @MakonnenMak. This one has been tricky in the past and I haven't been able to figure out what's actually going on. However looking closely it almost seems like it's trying to install with a path that doesn't exists. Are you able to try to manually set your `metals.javaHome` in your `:CocConfig` and try that? It will look something like this:

```
  "metals.javaHome": "/Users/ckipp/.sdkman/candidates/java/21.0.0.2.r11-grl",
```

**Reply by MakonnenMak** [reply]
Should've mentioned in my post (will update), but yeah I have that key set in the settings.
```
 {
           "metals.javaHome": "/usr/bin/java"
 }
```
Path confirmed via 
```
$ which java                                              [22:11:11]
/usr/bin/java
```

**Reply by MakonnenMak** [reply]
Completely forgot `which java` returns the symlink not the actual path. 
Updating it to `/usr/lib/jvm/java-15-openjdk` fixes the error.

Ran
`dirname $(dirname $(readlink -f $(which javac)))` to find the actual path just in case someone runs into this again.

Thanks for pointing out the issue @ckipp01 !
